core/src/main/java/org/mule/runtime/core/internal/metadata/NullMetadataResolverSupplier.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.metadata;
+
+import org.mule.runtime.extension.api.metadata.NullMetadataResolver;
+
+import java.util.function.Supplier;
+
+/**
+ * {@link Supplier} implementation which returns a {@link NullMetadataResolver} instance.
+ * This supplier always returns the same instance.
+ *
+ * @since 4.0
+ */
+public class NullMetadataResolverSupplier implements Supplier<NullMetadataResolver> {
```

Comment:
I understand, the problem that came up from using `Suppliers` is that you can't know if the resolver is a proper resolver or a `NullTypeResolver`.
**Why I need that?**
The logic of knowing if the type is dynamic or not is checking the type of the resolver, due that now is wrapped in a supplier that is not possible.
Which are the possible fixes?
* Instantiate the resolver and maintain the old logic (I don't like to instantiate a lot of classes just to ask that)
* Ask for the supplier type, having the `NullMetadataResolverSupplier` that will be returned every time that a `NullTypeResolver` fix the issue.